lib/experimental-default-theme.json

```diff
@@ -6,124 +6,148 @@
 				{
 					"name": "Black",
 					"slug": "black",
-					"color": "#000000"
+					"color": "#000000",
+					"origin": "core"
```

Comment:
It seems redundant to have "origin": "core", in all of the colors. Here they are defined in the core theme.json so our code should automatically assume all these colors of core origin.

Reply:
Yeah, thought about that. These are the trade-offs I've considered: 

1. We only write this once and themes don't have to, it's something internal to core so it's transparent to the consumers of the API.
2. If our code needs to augment that data, that needs to run for every request and makes the logic more complex.
3. We don't know how the UI control that uses this data is going to work.

With these trade-offs, I leaned towards implementing the simplest thing that works and that can scale as our needs evolve.

---

Alternatives I've considered:

- What if core colors (or colors not to show in the UI) were stored in a different part? Too complex to introduce at the stage we're in. We need to account for every level (top-level, blocks) and have access to both colors in the client for the case the theme doesn't provide any color.

- What if we add a `showInUi` flag instead of `origin`? Will work the same, not super-opinionated. Although I think `origin` is more future proof.

- What if we add the `origin` to all the colors? It's data we pass to the client that we don't need at the moment.

Reply:
Do we expect others to use this property? Something more descriptive like an array of which contexts to display the colors in UI may make sense.

My intuition would lean toward removing this property in the definition especially if we're unsure of usages by other consumers. Folks will copy paste examples, so I'd be a little careful here. We could do something like adding it in the in-memory representation or alternatively picking a different data structure like `"core": { "settings": { "color": "pallete" : {`. (I don't know this problem space very well, so just my quick takes, it may not make sense for this problem.)

Reply:
I've already shared my rationale about why I'd think going forward with the simplest approach that works is fine. However, one thing I didn't consider was the copy/paste rationale. That's a good point and makes me having second thoughts about this, to be honest.